src/Time/Bool.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE ExplicitForAll             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE InstanceSigs               #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeInType                 #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+module Time.Bool
+       ( And
+       , If
+       , type (==%)
+       ) where
+
+import Data.Type.Equality (type (==))
+
+import Time.Rational (type (:%), Rat)
+
+-- | Type-level 'if'.
+type family If (x :: Bool) a b where
+    If 'True  a _ = a
+    If 'False _ b = b
+
+-- | Type-level 'and'.
+type family And (a :: Bool) (b :: Bool) where
+  And 'False _ = 'False
+  And _      x = x
```

Comment:
What do you think about renaming `And` to `&&` operator? It might be more convenient to use (2 characters in infix usage instead of 5) but maybe this operator might look confusing... I think we need some package with type families for `Bool`. I didn't find such package on Hackage.